src/reconcile/rec_write.c

```diff
@@ -1580,18 +1584,6 @@ __rec_child_deleted(
 	}
 
 	/*
-	 * If there are deleted child pages that we can't discard immediately,
-	 * keep the page dirty so they are eventually freed.
-	 */
-	if (ref->addr != NULL) {
-		r->leave_dirty = 1;
-
-		/* This page cannot be evicted, quit now. */
-		if (F_ISSET(r, WT_EVICTING))
-			return (EBUSY);
-	}
-
-	/*
```

Comment:
I'm continuing to evict if the change is visible to the eviction thread, in other words, there's an older reader. That's probably wrong: in `rec_txn_read`, we switched to checking `wt_txn_committed` instead of "visibility" in the case of eviction.

I also dropped `r->leave_dirty`: I don't think that has any meaning for internal pages.